source/math/FloatPoint2DVectorList.ooc

```diff
@@ -107,4 +108,30 @@ FloatPoint2DVectorList: class extends VectorList<FloatPoint2D> {
 			result = result >> this[i] toString() >> "\n"
 		result
 	}
+	resampleLinear: func (start, end, interval: Float) -> This {
+		// Assumes list is sorted by x values.
+		resultCount := ((end - start) / interval) ceil() as Int + 1
+		result := This new()
+		previousIndex := 0
+		for (i in 0 .. resultCount) {
+			point := FloatPoint2D new(start + i * interval, 0.0f)
+			if (point x <= this[0] x)
+				point y = this[0] y
+			else if (point x >= this[this count - 1] x)
+				point y = this[this count - 1] y
+			else {
+				leftPoint, rightPoint: FloatPoint2D
+				for (j in previousIndex .. this count)
+					if (this[j] x <= point x) {
+						leftPoint = this[j]
+						rightPoint = this[Int minimum(j + 1, this count - 1)]
+						previousIndex = j
+					} else break
+				weight := Float absolute(point x - leftPoint x) / Float absolute(rightPoint x - leftPoint x)
```

Comment:
Ok, I double checked this pretty carefully to prove that I'm right! 
It is never checked explicitly, but since we are assuming that the list is sorted by x values, this case won't happen. `rightPoint` is always next in the list after `leftPoint`, so `rightPoint x > leftPoint x`. And the only way `leftPoint x > point x` is if `point x <= this[0] x` which is checked earlier. Also, `rightPoint x == leftPoint x` is never true even if there are duplicate points, because it will skip directly to the second point in the j loop. I will add a test with duplicate points though, that would be useful.